MoreLinq.Test/HasDuplicatesTest.cs

```diff
@@ -0,0 +1,99 @@
+namespace MoreLinq.Test
+{
+    using System.Collections.Generic;
+    using NUnit.Framework;
+
+    [TestFixture]
+    public class HasDuplicatesTest
+    {
+        [Test]
+        public void When_Asking_For_Duplicates_On_IEnumerable_Without_Duplicates_Then_False_Is_Returned()
+        {
+            var stringArray = new[]
+            {
+                "FirstElement",
+                "SecondElement",
+                "ThirdElement"
+            };
+
+            var hasDuplicates = stringArray.HasDuplicates();
+
+            Assert.That(hasDuplicates, Is.False);
+        }
+
+        [Test]
+        public void When_Asking_For_Duplicates_On_IEnumerable_With_Duplicates_Then_True_Is_Returned()
+        {
+            var stringArray = new[]
+            {
+                "FirstElement",
+                "DUPLICATED_STRING",
+                "DUPLICATED_STRING",
+                "ThirdElement"
+            };
+
+            var hasDuplicates = stringArray.HasDuplicates();
+
+            Assert.That(hasDuplicates, Is.True);
+        }
+
+        [Test]
+        public void When_Asking_For_Duplicates_On_IEnumerable_With_Custom_Always_True_Comparer_Then_True_Is_Returned()
+        {
+            var stringArray = new[]
+            {
+                "FirstElement",
+                "SecondElement",
+                "ThirdElement"
+            };
+
+            var hasDuplicates = stringArray.HasDuplicates(new DummyStringAlwaysTrueComparer());
+
+            Assert.That(hasDuplicates, Is.True);
+        }
+
+        [Test]
+        public void When_Asking_For_Duplicates_On_IEnumerable_With_Null_Comparer_Then_Default_Comparer_Is_Used_And_False_Is_Returned()
+        {
+            var stringArray = new[]
+            {
+                "FirstElement",
+                "SecondElement",
+                "ThirdElement"
+            };
+
+            var hasDuplicates = stringArray.HasDuplicates(null);
+
+            Assert.That(hasDuplicates, Is.False);
+        }
+
+        [Test]
+        public void When_Asking_For_Duplicates_On_IEnumerable_With_Custom_Always_False_Comparer_Then_False_Is_Returned()
```

Comment:
should be nice to add a test for opposite scenario: where duplicates exists, but the result is false because you passed a comparer that always returns false. 

also, I missed more 2 important tests:
- check that extension method is lazy (almost every morelinq method has this one)
- check that current implementation does not enumerate unnecessary elements (if you find a duplicate for example in the 3rd element, so you must not iterate the rest of elements)

Reply:
Good points! i'll add those after work :) 

Thanks for the feedback

Reply:
in my case, method isn't lazy since it returns a bool, but i still added a test for that!